ttnn/cpp/ttnn/operations/data_movement/untilize_with_unpadding/device/kernels/dataflow/writer_unary_unpad_width_16_sharded.cpp

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+#include "dataflow_api.h"
+
+// Special case writer for unpad width 16 tensors
+// Skip untilize and just copy f0 and f2 from input tiles to output tiles
+void kernel_main() {
```

Comment:
This is amazing! 
I think we should (outside the scope of this PR) talk about moving this kernel to a common directory so we can put this with other ops. 
I think a lot of ops can benefit from this optimized access pattern.